tests/forms_tests/tests/test_forms.py

```diff
@@ -746,6 +746,20 @@ class BeatleForm(Form):
             [str(bf[1]), str(bf[2]), str(bf[3])],
         )
 
+    def test_boundfield_index_with_non_int_non_slice(self):
+        class BeatleForm(Form):
+            name = ChoiceField(
+                choices=[('john', 'John'), ('paul', 'Paul'), ('george', 'George'), ('ringo', 'Ringo')],
+                widget=RadioSelect,
+            )
```

Comment:
`choices` and `widget` are unnecessary for this test. I think we can simplify it, e.g.
```python
def test_boundfield_index_with_non_int_non_slice(self):
    class TestForm(Form):
        name = ChoiceField(choices=[])

    msg = 'BoundField indices must be integers or slices, not str.'
    with self.assertRaisesMessage(TypeError, msg):
        TestForm()['name']['foo']
```

Reply:
Thanks, you're right. Simplified in the latest revision.

I've kept `TestForm()['name']` outside the context manager as I think it make sense to minimize the code inside the context to just what we expect (and wish to assert) raises a error.